### ISSUE TITLE
saltstack server_id changes with each run on python3

### ISSUE THREAD

**Issue by aboe76** [saltstack/salt#46595]
### Description of Issue/Question
saltstack server_id changes with each run

### Setup
Opensuse Tumbleweed 20180314
### Steps to Reproduce Issue
install tumbleweed, install salt-master, salt-minion, exchange keys,
run salt-call grains.item server_id multiple times
```
[root@spectre stack (master)]# salt-call grains.item server_id && date
local:
    ----------
    server_id:
        812924806
Sat Mar 17 14:05:41 CET 2018

[root@spectre stack (master)]# salt-call grains.item server_id && date
local:
    ----------
    server_id:
        533218382
Sat Mar 17 14:05:55 CET 2018

```

### Versions Report
```
Salt Version:
           Salt: 2018.1.99
 
Dependency Versions:
           cffi: 1.11.5
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: 0.26.0
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.26.0
         Python: 3.6.4 (default, Jan 03 2018, 13:52:55) [GCC]
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 17.0.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.3
 
System Versions:
           dist:   
         locale: UTF-8
        machine: x86_64
        release: 4.15.8-1-default
         system: Linux
        version: Not Installed
```

**Comment by aboe76** [comment]
Found the issue:

There are two changes in hash() function between Python 2.7 and Python 3.4
1. Adoptions of SipHash
2. Default enabling of Hash randomization

for more information
https://docs.python.org/3/using/cmdline.html#envvar-PYTHONHASHSEED

References:

Since from Python 3.4, it uses SipHash for it's hashing function. Read: https://lwn.net/Articles/574761/

Since Python 3.3 Hash randomization is enabled by default. Reference: object.__hash__ (last line of this section). Specifying PYTHONHASHSEED the value 0 will disable hash randomization. 

https://docs.python.org/3/reference/datamodel.html#object.__hash__

**Comment by Ch3LL** [comment]
ping @saltstack/team-suse any chance one of you can take a look here? Thanks

**Comment by Ch3LL** [comment]
oh nevermind I just realized this is also occurring outside of suse and can be replicated on arch.

**Comment by Ch3LL** [comment]
reading through the links you posted @aboe76 it seems the only way to ensure this is not randomized is by setting an environment variable: PYTHONHASHSEED

As the only purpose of this grain is to just return a integer ID based off the minion id I wonder if there is just a different approach we can take. ping @saltstack/team-core any ideas here on how to approach this?

**Comment by terminalmage** [comment]
Fixed in https://github.com/saltstack/salt/pull/46649.

**Comment by aboe76** [comment]
@terminalmage I have replaced core.py on my system with the new version from #46649 

I can confirm this works, needed to remove core.cpython-36.pyc from __pycache__
otherwise I still had the old core.py loading.

**Comment by terminalmage** [comment]
Thanks for confirming!

**Comment by terminalmage** [comment]
@aboe76 Keep in mind that this fix will go into the 2017.7.6 and 2018.3.1 releases, as we're nearing the release of 2018.3.0 and are not adding any more non-release-blocking fixes to that release.
